### DISCUSSION TITLE
Re-add soft line breaks to GitHub-flavored-markdown

### DISCUSSION THREAD

**Discussion by thislooksfun** [community/community#10981] [Other Feature Feedback, Questions, & Ideas]
Markdown supports soft line breaks, but it seems that GitHub flavored markdown has removed support for it. This is a real shame because it means I have to spend extra time re-formatting any github-destined markdown auto-generated markdown that is auto-generated from my commits (which have a hard line limit of 72 chars) because the resulting rendered text will have the same line breaks.

Here is an example block of text. This is formatted using a hard
72-character line limit, like I use for all my commits (and thus
auto-generated text like single-commit PR messages and semantic-release
changelogs).

Here is another block of text, but this time it's written as a single really long line. This is easy to type into the github text boxes, but that's not the only place text comes from. Why was support for soft-wrapping removed in the first place?

As you can see, the first block is *not* using soft-wrapping, despite each line ending in a single newline character (no spaces, no double newlines). I don't know if this is a bug or if it was an intentional change, but either way I would really appreciate it if it were changed back to allow standard markdown soft wrapping.

**Comment by mikepqr** [comment]
Agreed. Please restore [the behavior described in the docs](https://github.github.com/gfm/#soft-line-breaks)! (Or, failing that, fix the docs.)

**Reply by thislooksfun** [reply]
Unfortunately this behavior is not a violation of the docs :(

> A conforming parser may render a soft line break in HTML either as a line break or as a space.
> A renderer may also provide an option to render soft line breaks as hard line breaks.

**Reply by danielvoros-form3** [reply]
It seems to violate this part though:
> A regular line break (not in a code span or HTML tag) that is not preceded by two or more spaces or a backslash is parsed as a [softbreak](https://github.github.com/gfm/#softbreak).

Regular line breaks seem to be parsed as **hard breaks** and rendered as `<br />`

**Reply by thislooksfun** [reply]
Seems you did not read the qualification that appears immediately after that quoted section:

> A regular line break (not in a code span or HTML tag) that is not preceded by two or more spaces or a backslash is parsed as a [softbreak](https://github.github.com/gfm/#softbreak). (***A softbreak may be rendered in HTML either as a [line ending](https://github.github.com/gfm/#line-ending) or as a space.*** [emphasis mine] [...])

It's entirely in-spec for renderers to use `<br>` for softbreaks, even if I think it shouldn't be.

**Reply by liskin** [reply]
> It's entirely in-spec for renderers to use `<br>` for softbreaks, even if I think it shouldn't be.

I believe you're misinterpreting the spec. When it says "line ending", it means just a \n in the HTML, not a full `<br>`. They even link to the definition of a "line ending", which only talks about newline and carriage return control chars, not `<br>`.

So I don't think it's in-spec, actually.

**Reply by thislooksfun** [reply]
Ah, good point. However at the end of the section it does also say:

> A renderer may also provide an option to render soft line breaks as hard line breaks.

It seems that GitHub's renderer has that option and that the frontend is turning it on without letting us turn it off. Still technically within spec as far as I can tell, but I do wish it was up to the author whether it was a hard break.

After over two years of no fix though I've given up hope it'll be changed back. My main gripe (auto-populated PR descriptions) has been fixed anyway. Still would be nice though.

**Comment by nanmu42** [comment]
I just ran into this problem. All my `README.md` across many repos stops breaking lines as before, which is even obvious if there are some project badges.

![image](https://user-images.githubusercontent.com/8143068/166716179-197c5082-7ef0-48ea-b1f1-e82b79f3a4a1.png)

Is there a good reason to change this long-existed feature?

**Comment by stephenfuqua** [comment]
Wow. Typically I read my own markdown in my text editor, where having long lines of text is terrible. Just noticed the behavior in my repository, where I now have hard line breaks and... it looks terrible. This was a very surprising and distressing change. Please restore the old functionality.

Example of text that displayed at full width until recently:

![image](https://user-images.githubusercontent.com/9324390/166720342-502082eb-e6be-4b26-9c3d-4472d6c9e385.png)

https://github.com/Ed-Fi-Exchange-OSS/Suite-3-Performance-Testing/blob/main/README.md

**Reply by jfmc** [reply]
Same here. Virtually *ALL* readme in all repositories are broken now. I've written a markdown-based documentation system that is also broken (so I suppose that many other markdown render code will be). I used to edit .md files in emacs and do M-q to reflow paragraphs. This is no longer possible now. If this is not a bug, this is a HORRIBLE idea.

**Reply by stephenfuqua** [reply]
In the past few minutes, rendering has reverted back to the previous (desirable) behavior. Thank you, GitHub team.

![image](https://user-images.githubusercontent.com/9324390/166725557-37583103-7b0e-49c0-a640-d604ad707fc9.png)

**Reply by fshanlee** [reply]
I just created some markdown in a text editor, which I copied and pasted into my repository. I was disappointed to note that hard line breaks appear throughout the rendered output, so I Googled and found this discussion. It seems that the rendering reverted back to the previous, desired behaviour in 2022, but that's not what I am observing now in my repository. @stephenfuqua , have you noticed any change in your repository?

**Reply by rmacklin** [reply]
It should work fine in markdown files inside a repository - where it doesn't work is in comment fields/issue descriptions

**Reply by fshanlee** [reply]
@rmacklin , thank you very much for clarifying - my markdown was in an issue description. I just created a test file within the actual repository, with exactly the same markdown, and this time it renders correctly. Thanks again for your help.

**Comment by wcerfgba** [comment]
Different repos have different needs. I want hard line breaks because I write poetry, and don't want to add `  ` to the end of every line, or `\` or `<br/>`. An option to choose would be best. Also some way to do the same in Jekyll for GitHub Pages.

**Comment by mickvangelderen** [comment]
Was the change also reverted for GitHub releases? It seems like the release notes still render every line-break, even within a markdown paragraph.

**Comment by deminimis** [comment]
Would be nice if enter was a hard line break, and shift+enter was what it currently was. Would be faster than two spaces.

**Comment by aymena80s545** [comment]
https://t.me/Helloo_FRIENDD
